ssl-credential-creator/ssl_credential_creator.py

```diff
@@ -0,0 +1,74 @@
+import random
+from OpenSSL import crypto, SSL
+
+# File names for generated credentials
+ROOT_CA_PRIV_KEY_FILE = "root_ca_priv_key.key"
+ROOT_CA_CERT_FILE = "root_ca_cert.crt"
+SERVER_PRIV_KEY_FILE = "server_priv_key.key"
+SERVER_CERT_FILE = "server_cert.crt"
+DEVICE_PRIV_KEY_FILE = "device_priv_key.key"
+DEVICE_CERT_FILE = "device_cert.crt"
+
+# Reusable certificate subject for testing
+TEST_CERT_SUBJECT = crypto.X509Name(crypto.X509().get_subject())
+TEST_CERT_SUBJECT.C = 'US'
+TEST_CERT_SUBJECT.ST = 'Test'
+TEST_CERT_SUBJECT.L = 'Test'
+TEST_CERT_SUBJECT.O = 'Test'
+TEST_CERT_SUBJECT.OU = 'Test'
+TEST_CERT_SUBJECT.CN = 'localhost'
+TEST_CERT_SUBJECT.emailAddress = 'test@test.com'

```

Comment:
Think it might make sense to replace the `Test` values about with `Test_<VALUE>`. I.e.  `TEST_CERT_SUBJECT.ST = 'Test'`-> `TEST_CERT_SUBJECT.ST = 'Test_STATE'` for future debugging

Reply:
I can do this for some of the fields but not all - namely the common name as it is typically checked against when doing TLS handshakes.